### PR TITLE
Salvage old contributor docs

### LINKED ISSUE #4003: Salvage content from old contributor docs
There is good content left in help/contributor. There is also a lot of content about Hugo that no longer applies.

1. Delete obsolete content within help/contributor
1. Update remaining content in help/contributor (delete Hugo front matter, delete broken links, rename files as needed, etc)
1. Copy the remaining content in help/contributor to docs/help/contributor
1. Delete help/contributor

### PR DESCRIPTION
**Pull request by RichardJJG**
Fixes #4003

### CODE REVIEW

**Comment by RichardJJG** [comment]
I know there are some instances of "[insert link here]" -- I did a lot of moving and deleting files and didn't want to run foul of the prow-bot. We can insert these links after the merge.

File: docs/help/contributor/github.md [hunk 1]
```diff
@@ -1,9 +1,4 @@
----
-title: "GitHub workflow for Knative documentation"
-linkTitle: "Using GitHub"
-weight: 10
-type: "docs"
----
+# GitHub workflow for Knative documentation
```
Comment (julz) [inline docs/help/contributor/github.md:1]: looks like some of the syntax in this file doesn't work with mkdocs (specifically the `## Set up your local machine{#clone}` bit - mkdocs doesn't have `{#clone}` to set up an anchor id, instead the id is generated based on the header text (https://www.mkdocs.org/user-guide/writing-your-docs/#linking-to-pages)

Comment (RichardJJG) [reply docs/help/contributor/github.md:1]: Thank you; well-spotted.

File: docs/help/contributor/creating-blog-posts.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ type: "docs"
 showlandingtoc: "false"
 ---
```
Comment (abrennan89) [inline docs/help/contributor/creating-blog-posts.md:8]: Remove frontmatter

File: docs/help/contributor/creating-blog-posts.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ type: "docs"
...
 
+# Creating blog posts
+
 The Knative blog is owned by the [Documentation working group](https://knative.dev/community/contributing/working-groups/working-groups/#documentation) and run by the [Editorial Team](#leadership).
```
Comment (abrennan89) [inline docs/help/contributor/creating-blog-posts.md:11]: Editorial team?

Comment (abrennan89) [reply docs/help/contributor/creating-blog-posts.md:11]: Also should this be DUX WG now?

File: docs/help/contributor/creating-new-docs.md [hunk 4]
```diff
@@ -1,26 +1,17 @@
...
+# Creating new documentation
 
 To contribute new documentation, follow these steps:
 
 1. Identify the audience and intended use for the information.
 1. Choose the [type of content](#content-types) you wish to contribute.
-1. You can use a template to get started:
-   * [New docs file](./docs-page.md)
-   * [New `_index.md` file](./index-page.md)
+<!-- Add "Choose a template" step with links to all the templates. -->
```
Comment (abrennan89) [inline docs/help/contributor/creating-new-docs.md:7]: Create a follow up issue for this if it's TBA?

File: docs/help/contributor/creating-new-docs.md [hunk 5]
```diff
@@ -1,26 +1,17 @@
...
 1. [Choose appropriate titles and filenames](#choosing-titles-and-filenames).
-1. Write your new content. See the [How-to guides](/help/contributor/how-to/)
-   to help you with this process. Feel free to reach out to the
-   [Docs Working Group](/help/contributor/gettingstarted.md#get-help-from-the-community)
-   with any questions.
+1. Write your new content. See the style guide to help you with this process.
```
Comment (abrennan89) [inline docs/help/contributor/creating-new-docs.md:9]: Add link to the style guide?

File: docs/help/contributor/creating-new-docs.md [hunk 6]
```diff
@@ -1,26 +1,17 @@
...
+   Feel free to reach out to the
+   [Docs Working Group](/becoming-a-contributor.md#get-help-from-the-community) with any questions.
```
Comment (abrennan89) [inline docs/help/contributor/creating-new-docs.md:11]: DUX?

File: docs/help/contributor/creating-new-docs.md [hunk 7]
```diff
@@ -31,7 +22,7 @@ structure, or the necessary supporting information. The following examples show
 this principle in action:
 
 - The reader needs to perform a specific task: Tell them how to recognize when
```
Comment (abrennan89) [inline docs/help/contributor/creating-new-docs.md:24]:
```suggestion
- The reader needs to perform a specific task. Tell them how to recognize when
```

File: docs/help/contributor/creating-new-docs.md [hunk 8]
```diff
@@ -31,7 +22,7 @@ structure, or the necessary supporting information. The following examples show
...
-  the task is necessary and provide the task itself as a list of numbered steps,
+  the task is necessary and provide the task itself as a list of numbered steps;
   don’t simply describe the task in general terms.
```
Comment (abrennan89) [inline docs/help/contributor/creating-new-docs.md:26]:
```suggestion
  the task is necessary, and provide the task itself as a list of numbered steps. Don’t simply describe the task in general terms.
```

Comment (abrennan89) [reply docs/help/contributor/creating-new-docs.md:26]: suggestion for readability - not sure the last sentence is even needed here tbh

**Review by abrennan89 — CHANGES_REQUESTED** [review]
some comments / suggestions

**Comment by RichardJJG** [comment]
All suggestions acted on, thanks, @abrennan89. I moved the templates to docs/help/contributor/templates -- it makes more sense for them to live there than cluttering up the top level.

**Review by abrennan89 — APPROVED** [review]
Awesome, thanks @RichardJJG ! 🙂 
/lgtm

**Pull request merged**